test/e2e/deploy/operator.go

```diff
@@ -56,8 +56,12 @@ func CreateDeploymentCR(f *framework.Framework, dep *unstructured.Unstructured)
 }
 
 func DeleteDeploymentCR(f *framework.Framework, name string) {
+	// Delete all
+	deletionPolicy := metav1.DeletePropagationForeground
```

Comment:
FWIW, in deploy.go I had intentionally written the code so that `RemoveObject` loops until all objects are truly gone.

Using `metav1.DeletePropagationForeground` here should have the same effect and makes sense because we are deleting just one object; I'm just wondering whether we can 100% rely on it or also should wait for the "not found error".

Reply:
Ok, I added that check too, to wait till we get "not found" error.